### PR TITLE
Fix API rule violations

### PR DESCRIPTION
**Pull request by Avni-Sharma**
Fix issue #373

### CODE REVIEW

File: deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml [hunk 1]
```diff
@@ -259,7 +259,7 @@ spec:
         status:
           description: ServiceBindingRequestStatus defines the observed state of ServiceBindingRequest
           properties:
-            applications:
+            applicationObjects:
```
Comment (sbose78) [inline deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml]: How about flipping this, call everything `applications`?

**Comment by Avni-Sharma** [comment]
Currently, we have 
```
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [apps:[v1alpha1], ] 
API rule violation: list_type_missing,github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1,ServiceBindingRequestStatus,Applications
API rule violation: list_type_missing,github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1,ServiceBindingRequestStatus,Conditions
```

To solve this we will have to do
```
Conditions *[]conditionsv1.Condition `json:"conditions"`

```

However, Ideally, we should not have `*[]conditionsv1` . I believe we can ignore such violations because, on a quick check with upstream operators, I found that `[]conditionsv1` is being used.

**Comment by sbose78** [comment]
@Avni-Sharma Can I take this in?

**Comment by Avni-Sharma** [comment]
> @Avni-Sharma Can I take this in?

Yes

**Pull request merged**